src/DynamoCore/Nodes/NodeCategories.cs

```diff
@@ -163,6 +166,88 @@ public static string PreprocessTypeName(string fullyQualifiedName)
         }
 
         /// <summary>
+        /// This method returns a name for the icon based on name of the node.
+        /// </summary>
+        /// <param name="descriptor">Function descriptor, that contains all info about node.</param>
+        /// <param name="overridePrefix">
+        /// overridePrefix is used as default value for generating node icon name.
+        /// If overridePrefix is empty, it uses QualifiedName property.
+        /// e.g. Autodesk.DesignScript.Geometry.CoordinateSystem.ByOrigin
+        /// </param>
```

Comment:
I have added [an internal task](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6125) so that we don't forget to create unit test cases for `TypedParametersToString` (maybe there are already tests, if so please add them in this PR and close the internal task). Thanks!